[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict.

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89981
* #89980
* #89900
* #89899
* #89898

The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.

fegin committed:
[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict.

[ghstack-poisoned]

awgu reviewed (approved):
LGTM! Thanks for cleaning this up :)

fegin committed:
Update on "[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict."


The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.


[ghstack-poisoned]

fegin committed:
Update on "[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict."


The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.


[ghstack-poisoned]

fegin committed:
Update on "[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict."


The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.


[ghstack-poisoned]

fegin committed:
Update on "[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict."


The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.


[ghstack-poisoned]

fegin committed:
Update on "[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict."


The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.


[ghstack-poisoned]

fegin committed:
Update on "[FSDP][optim_state_dict][5/N] Remove optim_inputs for sharded state_dict."


The argument, `optim_inputs`, is being deprecated. Sharded optimizer state_dict APIs are not be used. It is safe to remove them.


[ghstack-poisoned]
